src/bitmessagemain.py

```diff
@@ -218,17 +236,34 @@ def start(self):
                 sys.exit()
             elif opt in ("-d", "--daemon"):
                 daemon = True
-                state.enableGUI = False # run without a UI
+                state.enableGUI = False  # run without a UI
             elif opt in ("-c", "--curses"):
                 state.curses = True
             elif opt in ("-t", "--test"):
-                state.testmode = daemon = True
-                state.enableGUI = False # run without a UI
+                state.testmode = True
+                if os.path.isfile(os.path.join(
+                        state.appdata, 'unittest.lock')):
+                    daemon = True
+                state.enableGUI = False  # run without a UI
+                # Fallback: in case when no api command was issued
+                state.last_api_response = datetime.now()
+                # Apply special settings
+                config = BMConfigParser()
+                config.set(
```

Comment:
`
CONFIGURATIONS = [('bitmessagesettings', 'apienabled', ('bitmessagesettings', 'apiusername', 'username'), ('bitmessagesettings', 'apipassword', 'password'), ('bitmessagesettings', 'apinotifypath', os.path.join(app_dir, 'tests', 'apinotify_handler.py')]`

for data in CONFIGURATIONS:
                                 config.set(data)

Reply:
Any update here?

Reply:
This looks ugly

Reply:
Hmm not sure whether this is necessary, I think it can be kept as it is. It can always be changed later.

Reply:
@PeterSurda, I thought of moving this into `testConfig` when it will load a sort of fixtures and merge them into `keys.dat` and then run bitmessage and check results. But for now it's too complex. I suspect this PR will be merged never.